core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -1799,6 +1799,43 @@ public void testExpireSnapshotsWithExecutor() {
         .isGreaterThan(0);
   }
 
+  @TestTemplate
+  public void testRemoveMetadataWithNoSnapshots() throws Exception {
+    TableMetadata current = table.ops().current();
+    removeSnapshots(table)
+        .expireOlderThan(System.currentTimeMillis())
+        .retainLast(1)
+        .cleanExpiredMetadata(true)
+        .commit();
+    assertThat(table.ops().current())
+        .as("No snapshot or metadata to remove, should be a no-op")
+        .isSameAs(current);
```

Comment:
@lirui-apache I think I'd prefer to pull this out to a separate test `testRemoveSnapshotsNoOp()`, it's generally better to keep tests focused on a single aspect we're trying to test rather than have multiple assertions for different cases